examples/weather/diagnostic/diagnostic/models.py

```diff
@@ -0,0 +1,47 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union
+
+from modulus import Module
+from modulus.models.afno import AFNO
+
+
+default_model_params = {
+    "afno": {
+        "patch_size": (8, 8),
+        "embed_dim": 768,
+        "depth": 12,
+        "num_blocks": 8,
+    }
+}
+
+
+def setup_model(
+    model_type: str = "afno", model_name: Union[str, None] = None, **model_cfg
+) -> Module:
+    """Setup model from config dict."""
+    model_kwargs = default_model_params[model_type].copy()
+    model_kwargs.update(model_cfg)
+
+    if model_type == "afno":
```

Comment:
We can actually use the `instantiate` Module method. There is an example of it here, https://github.com/loliverhennigh/modulus/blob/fea-unified_recipe_model_card/examples/weather/unified_recipe/train.py#L99. This will make instantiating the model automatic from configs.

Reply:
Great suggestion, changed in [2e2f4b7](https://github.com/NVIDIA/modulus/pull/356/commits/2e2f4b7f46350d2db3bc801e09ca2befa9a1878c). This way you can use any compatible model without having to add it to the list.